src/main/java/ch/uzh/ifi/hase/soprafs22/controller/GameController.java

```diff
@@ -32,8 +43,20 @@ public class GameController {
     public void unitAttack(@RequestHeader("token") String token, @PathVariable Long id, @RequestBody UnitCommandPutDTO unitCommandPutDTO) {
         Position attacker = DTOMapper.INSTANCE.convertPositionDTOToPosition(unitCommandPutDTO.getStart());
         Position defender = DTOMapper.INSTANCE.convertPositionDTOToPosition(unitCommandPutDTO.getEnd());
+        Unit defendingUnit = gameService.unitAttack(id, token, attacker, defender);
 
-        gameService.unitAttack(id, token, attacker, defender);
+        try {
```

Comment:
we shouldn't "try and catch" in a controller

if you do it like @luis-tm below proposed we won't need that

Reply:
@luis-tm way looks cleaner, but why should't we try/catch in controller?

Now it is is just hidden in the framework (https://github.com/spring-projects/spring-framework/blob/061d1263f854a132d76f3528034f907107c1727e/spring-messaging/src/main/java/org/springframework/messaging/core/AbstractMessageSendingTemplate.java#L186)